src/shared/dissect-image.c

```diff
@@ -1707,6 +1708,12 @@ static int run_fsck(int node_fd, const char *fstype) {
                 return 0;
         }
 
+        r = find_executable("fsck", &fsck_path);
+        if (r < 0) {
+                log_debug("Unexpectedly didn't find fsck binary itself.");
+                return 0;
```

Comment:
shouldn't this return an error?

Reply:
L1708 does `return 0` in case the `fsck.$fstype` doesn't exist, I guess it's best to mimic that behaviour if the `fsck` binary itself couldn't be found too.

Reply:
We proceed anyway if can't determine whether fsck binary for some specific fstype exists, but the lack of main fsck binary should be considered as an error. (see below, we exec() /sbin/fsck unconditionally before)

Also, better to rephrase it as "Cannot find fsck binary"